app/models/run.rb

```diff
@@ -53,9 +57,23 @@ def program(string_or_symbol)
       h[string_or_symbol.to_s]
     end
 
-    def program_from_content_type(content_type_string)
-      Run.exportable_programs.each do |prg|
-        return prg if prg.content_type == content_type_string
+    def program_from_attribute(func_name, value)
+      Run.programs.map do |prg|
+        case func_name
+        when 'to_s'
+          prg_value = prg.to_s
+        when 'to_sym'
+          prg_value = prg.to_sym
+        when 'file_extension'
+          prg_value = prg.file_extension
+        when 'website'
+          prg_value = prg.website
+        when 'content_type'
+          prg_value = prg.content_type
+        when 'exportable'
+          prg_value = prg.exportable
+        end
+        return prg if prg_value == value
       end
       return nil
```

Comment:
this `return nil` case should be a super obvious and unmissable error, like a `raise`, because it means there is a consumer calling this function incorrectly